File: src/main/java/ru/TlkHeader.java
package ru;

import java.io.IOException;
import java.io.InputStream;

public class TlkHeader {
    public int magic;
    public int ver;
    public int min_ver;
    public int entry1Count;
    public int entry2Count;
    public int treeNodeCount;
    public int dataLen;

    public TlkHeader(InputStream r) throws IOException {
        this.magic = TlkFile.readInt32(r);
        this.ver = TlkFile.readInt32(r);
        this.min_ver = TlkFile.readInt32(r);
        this.entry1Count = TlkFile.readInt32(r);
        this.entry2Count = TlkFile.readInt32(r);
        this.treeNodeCount = TlkFile.readInt32(r);
        this.dataLen = TlkFile.readInt32(r);
    }

}

File: python_ver/tlk_header.py
import input_stream
import tlk_file


class TlkHeader:
    def __init__(self, in_stream):
        self.magic = in_stream.read_int_32()  # TODO ???
        self.ver = in_stream.read_int_32()  # TODO ???
        self.min_ver = in_stream.read_int_32()  # TODO ???
        self.entry_1_count = in_stream.read_int_32()  # TODO ???
        self.entry_2_count = in_stream.read_int_32()  # TODO ???
        self.tree_nodes_count = in_stream.read_int_32()  # count of unique symbols
        self.data_len = in_stream.read_int_32()  # count of bytes of encoded sequence
